### PR TITLE
updated wireguard docs for crux branch

### PR DESCRIPTION
**Pull request by usman-umer**
Same changes made in my previous PR in the master branch, but for the crux branch instead.

### CODE REVIEW

File: docs/configuration/interfaces/wireguard.rst [hunk 1]
```diff
@@ -1,148 +1,406 @@
...
 
-The public key is being shared with your peer(s), your peer will encrypt all
-traffic to your system using this public key.
+.. opcmd:: generate wireguard default-keypair
+
+   It generates the keypair, which includes the public and private parts,
+   and stores it within VyOS. It will be used per default on any configured
+   WireGuard interface, even if multiple interfaces are being configured.
+
+.. opcmd:: show wireguard keypairs pubkey default
```
Comment (rebortg) [inline docs/configuration/interfaces/wireguard.rst:42]: the command in crux is `show wireguard pubkey`

File: docs/configuration/interfaces/wireguard.rst [hunk 2]
```diff
@@ -1,148 +1,406 @@
...
+
+   It shows the public key to be shared with your peer(s). Your peer will
+   encrypt all traffic to your system using this public key.
+
+   .. code-block:: none
+
+     vyos@vyos:~$ show wireguard keypairs pubkey default
```
Comment (rebortg) [inline docs/configuration/interfaces/wireguard.rst:49]: `show wireguard pubkey`

File: docs/configuration/interfaces/wireguard.rst [hunk 3]
```diff
@@ -1,148 +1,406 @@
...
+own keypairs.
 
-The next step is to configure your local side as well as the policy based
-trusted destination addresses. If you only initiate a connection, the listen
-port and endpoint is optional, if you however act as a server and endpoints
-initiate the connections to your system, you need to define a port your clients
-can connect to, otherwise it's randomly chosen and may make it difficult with
-firewall rules, since the port may be a different one when you reboot your
-system.
+.. opcmd:: generate wireguard named-keypairs <name>
```
Comment (rebortg) [inline docs/configuration/interfaces/wireguard.rst:60]: `named-keypairs` are not available in crux

File: docs/configuration/interfaces/wireguard.rst [hunk 4]
```diff
@@ -1,148 +1,406 @@
...
+
+**local side - commands**
 
 .. code-block:: none
 
-  set interfaces wireguard wg01 address '10.1.0.1/24'
+  set interfaces wireguard wg01 address '10.1.0.1/30'
   set interfaces wireguard wg01 description 'VPN-to-wg02'
-  set interfaces wireguard wg01 peer to-wg02 allowed-ips '10.2.0.0/24'
-  set interfaces wireguard wg01 peer to-wg02 endpoint '192.168.0.142:12345'
```
Comment (rebortg) [inline docs/configuration/interfaces/wireguard.rst:59]: this is the correct syntax for the endpoint configuration in crux
`address` and `port` are not available in crux

File: docs/configuration/interfaces/wireguard.rst [hunk 5]
```diff
@@ -1,148 +1,406 @@
...
+through the WireGuard interface `wg01`. Multiple IPs or networks can be
+defined and routed. The last check is allowed-ips which either prevents
+or allows the traffic.
 
-**remote side**
+.. note:: You can not assign the same allowed-ips statement to multiple
+   WireGuard peers. This a a design decision. For more information please
+   check the `WireGuard mailing list`_.
+
+.. cfgcmd:: set interfaces wireguard <interface> private-key <name>
```
Comment (rebortg) [inline docs/configuration/interfaces/wireguard.rst:122]: this option is not available in crux

File: docs/configuration/interfaces/wireguard.rst [hunk 6]
```diff
@@ -1,148 +1,406 @@
...
+Some users tend to connect their mobile devices using WireGuard to their VyOS
+router. To ease deployment one can generate a "per mobile" configuration from
+the VyOS CLI.
+
+.. warning:: From a security perspective it is not recommended to let a third
+  party create and share the private key for a secured connection. You should create the
+  private portion on your own and only hand out the public key. Please keep this
+  in mind when using this convenience feature.
+
+.. opcmd:: generate wireguard client-config <name> interface <interface> server <ip|fqdn> address <client-ip>
```
Comment (rebortg) [inline docs/configuration/interfaces/wireguard.rst:385]: this is also not available in crux

File: docs/configuration/interfaces/wireguard.rst [hunk 7]
```diff
@@ -1,148 +1,406 @@
...
 
-.. code-block:: none
+********************
+Operational Commands
+********************
+
+Status
+======
+
+.. opcmd:: show interfaces wireguard wg0 summary
```
Comment (rebortg) [inline docs/configuration/interfaces/wireguard.rst:290]: command not available in crux

**Review by rebortg — CHANGES_REQUESTED** [review]
i added some hints, crux have in some configs a different syntax.

**Comment by usman-umer** [comment]
Modified commands for crux.

**Comment by rebortg** [comment]
sorry, there are still the wrong command in multiple locations.

this, for example, is wrong in crux:
`set interfaces wireguard wg01 peer to-wg02 address '<Site2 Pub IP>' `
`set interfaces wireguard wg01 peer to-wg02 port '58120'`

replace it with something like this:
`set interfaces wireguard wg01 peer to-wg02 endpoint '192.168.0.142:12345'`

**Comment by usman-umer** [comment]
Replaced interface address with endpoint & ports.

**Pull request merged**